### PR TITLE
EASYOPAC-720 - New campaign widget 'Grid campaign'.

### PR DESCRIPTION
**Pull request by bozeak**

### CODE REVIEW

File: modules/ding_campaign/plugins/content_types/grid_campaign.inc [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+  $block->content = ding_campaign_display($context_data, $count, $offset, $style, $subtype, $more_link);
+
+  return $block;
+}
+
+/**
+ * Implements hook_content_type_edit_form().
+ */
+function ding_campaign_grid_campaign_content_type_edit_form($form, &$form_state) {
+
```
Comment (NicolaeGrigoriu) [inline modules/ding_campaign/plugins/content_types/grid_campaign.inc]: Remove empty line

File: modules/ding_campaign/plugins/content_types/grid_campaign.inc [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+  $form['ding_campaign_count'] = array(
+    '#type' => 'textfield',
+    '#title' => t('Campaign count'),
+    '#description' => t('The maximum number of campaigns to display'),
+    '#default_value' => (isset($form_state['conf']['ding_campaign_count'])) ? $form_state['conf']['ding_campaign_count'] : 3,
+    '#size' => 2,
+    '#maxlength' => 2,
```
Comment (NicolaeGrigoriu) [inline modules/ding_campaign/plugins/content_types/grid_campaign.inc]: For validation consider [element_validate_integer_positive](https://api.drupal.org/api/drupal/includes%21form.inc/function/element_validate_integer_positive/7.x)

File: modules/ding_campaign/plugins/content_types/grid_campaign.inc [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+  );
+
+  $form['ding_campaign_offset'] = array(
+    '#type' => 'textfield',
+    '#title' => t('Campaign offset'),
+    '#description' => t('Which campaign should we count from when returning campaigns. This is used to avoid displaying the same campaigns when using multiple campaign panes on the same panel. Use 0 to return the most relevant campaigns.'),
+    '#default_value' => (isset($form_state['conf']['ding_campaign_offset'])) ? $form_state['conf']['ding_campaign_offset'] : 0,
+    '#size' => 3,
+    '#maxlength' => 3,
```
Comment (NicolaeGrigoriu) [inline modules/ding_campaign/plugins/content_types/grid_campaign.inc]: For validation consider [element_validate_integer_positive](https://api.drupal.org/api/drupal/includes%21form.inc/function/element_validate_integer_positive/7.x)

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: c7b124d22c5f97b732e6fedc08dde89051e8514d
  * Commits must be in the following format: [issue ID] - [short summary of change]

Guidelines are available at https://github.com/easySuite/ding2

---
This message was auto-generated by https://gitcop.com

**Comment by guddo** [comment]
@NicolaeGrigoriu Why do you merge when **gitcop** and **Scrutinizer** is complaining?

**Comment by NicolaeGrigoriu** [comment]
@guddo scrutinizer issues are related to a type hint in hook this one is not necessary a code style issue.

**Comment by guddo** [comment]
@NicolaeGrigoriu If it does pass Scrutinizer it doesn't follow the Code Guidelines. What about the gitcop?

**Comment by bozeak** [comment]
@guddo Gitcop notice appeared because of incorrect commit message in which I unfortunately forgot to add issue ID match correct format. Is that a great problem?

**Comment by guddo** [comment]
@bozeak I understand, pure simple. It is a problem that comments in Pull Requests are ignored by Code Reviewer before PR is approved and Merged.

**Comment by NicolaeGrigoriu** [comment]
https://images.gr-assets.com/hostedimages/1392607079ra/8583647.gif

**Pull request merged**